src/Graph/GraphPrinter.php

```diff
@@ -174,13 +174,28 @@ protected function processResultRow( array /* of SMWResultArray */ $row ) {
 			while ( ( /* SMWWikiPageValue */
 				$object = $resultArray->getNextDataValue() ) !== false ) {
 
-				// create SRF\GraphNode for column 0
-				if ( $i == 0 ) {
-					$node = new GraphNode( $object->getShortWikiText() );
+				$id = $object->getShortWikiText();
+				$nodeExists = false;
+
+				// check, if node has already been created
+				/** @var GraphNode $node */
+				foreach ( $this->nodes as $node ) {
+					if ( $node->getID() == $id ) {
+						$nodeExists = true;
+						break;
+					}
+				}
+
+				if ( !$nodeExists ) {
```

Comment:
```
if ( !$this->nodeExists( $id ) ) {
```

Which itself could go in a 

```
$this->newGrapthNode( $object->getShortWikiText() )
```

Reply:
Sorry, @JeroenDeDauw what do you mean with your comment?

Reply:
any progress here? @JeroenDeDauw @gesinn-it-gea ?